dojo/tools/semgrep/models.py

```diff
@@ -0,0 +1,56 @@
+
+
+class SemgrepJSONResult:
+
+    def __init__(self, extra=None, path='', start=None, end=None):
+        self.path = path
+
+        self.start = 0
+        self.end = 0
+        self.severity = "Info"
+        self.title = "SemGrep detection rule"
+        self.message = "Detected by semgrep rule"
+        self.fix = "None"
+        self.lines = "None"
+        self.references = "Detected by semgrep rule"
+        self.test = "SemGrep rule"
+        self.description = "Detected by semgrep rules"
+        self.cwe = 0
+
+        if start is not None:
+            self.start = start['line']
+        if end is not None:
+            self.end = end['line']
+
+        if extra is None:
+            return
+
+        metadata, metavars = extra.get('metadata'), extra.get('metavars')
+        self.fix = extra.get('fix')
+        self.lines = extra.get('lines')
+        self.message = extra.get('message')
+
+        if not metadata:
+            return
+
+        # parse CWE
+        self.title = metadata.get("cwe").partition(':')[2]
```

Comment:
hey I'm one of the maintainers of semgrep and I'm excited that DefectDojo is integrating semgrep. Happy to help if needed with things. I noticed that this line will error out if `metadata.cwe` is not defined and will throw 

```
AttributeError: 'NoneType' object has no attribute 'partition'
```

perhaps change to `metadata.get("cwe", "")`?

Reply:
Yeah, I guess that makes sense. I was telling Sebastien he has adapted some of my code in this PR off of a fork of my fork it seems, and I just saw the emails now. I was not splitting on colon at all, so I did not even notice, and I was not sure yet from playing which fields in the `"metadata": {}` struct are optional and mandatory from looking at semgrep code (I am OK with Python, less so with OCaml).

But I defer to the majority opinion here. I worked on this to force myself to start evaluating semgrep more seriously, where as Sebastian is actually using it and wants it in DD, so his opinions are probably way more informed than mine!

Reply:
Will make a patch tomorrow. This seems to be very simple to handle @DrewDennison comment

Reply:
done